controllers/manilaapi_controller.go

```diff
@@ -638,33 +638,45 @@ func (r *ManilaAPIReconciler) reconcileNormal(ctx context.Context, instance *man
 	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	ctrlResult, err := r.getSecret(ctx, helper, instance, instance.Spec.TransportURLSecret, &configVars)
-	if err != nil {
+
+	ctrlResult, err := getServiceSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.Secret},
+		[]string{
+			instance.Spec.PasswordSelectors.Service,
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		manila.NormalDuration,
+		&configVars,
+	)
+	if (err != nil || ctrlResult != ctrl.Result{}) {
 		return ctrlResult, err
 	}
 
 	//
-	// check for required service secrets
+	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	for _, secretName := range instance.Spec.CustomServiceConfigSecrets {
-		ctrlResult, err = r.getSecret(ctx, helper, instance, secretName, &configVars)
-		if err != nil {
-			return ctrlResult, err
-		}
-	}
 
 	parentManilaName := manila.GetOwningManilaName(instance)
-
-	parentSecrets := []string{
+	secretNames := []string{
```

Comment:
@ASBishop the idea here is to collect the `secretNames` beforehand, including any potential secret referenced by `customServiceConfigSecret`, and make a single call to `getConfigSecret`